File: backend/src/main/java/iuh/labbooking/service/attendance/AttendanceService.java
package iuh.labbooking.service.attendance;

import iuh.labbooking.dto.request.attendance.CheckInRequest;
import iuh.labbooking.dto.request.attendance.CheckOutRequest;
import iuh.labbooking.dto.response.attendance.AttendanceResponse;
import iuh.labbooking.dto.response.attendance.AttendanceStatusResponse;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface AttendanceService {
    
    AttendanceResponse checkIn(Long bookingId, CheckInRequest request);
    
    AttendanceResponse checkOut(Long bookingId, CheckOutRequest request);

    AttendanceStatusResponse findAttendanceStatusByBookingId(Long bookingId);

    List<AttendanceResponse> findAttendancesByBookingId(Long bookingId);
}

File: frontend/src/services/attendance.service.ts
import http from '@/lib/http'
import type { ApiResponse } from '@/schemas/base.schema'
import type {
  CheckInRequest,
  CheckOutRequest,
  AttendanceResponse,
  AttendanceStatusResponse
} from '@/schemas/booking-attendance.schema'

const attendanceService = {
  checkIn: (bookingId: number, body: CheckInRequest) =>
    http.post<ApiResponse<AttendanceResponse>>(`/attendances/${bookingId}/checkin`, body),

  checkOut: (bookingId: number, body: CheckOutRequest) =>
    http.post<ApiResponse<AttendanceResponse>>(`/attendances/${bookingId}/checkout`, body),

  getStatus: (bookingId: number) => http.get<ApiResponse<AttendanceStatusResponse>>(`/attendances/status/${bookingId}`),

  getByBookingId: (bookingId: number) =>
    http.get<ApiResponse<AttendanceResponse[]>>(`/attendances/booking/${bookingId}`)
}

export default attendanceService
